### PR TITLE
support both types

### PR DESCRIPTION
**Pull request by MixailE**

### CODE REVIEW

File: src/helpers/getAllowList.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { KetlAllowMap__factory } from '@big-whale-labs/ketl-allow-map-contract'
+import defaultProvider from 'helpers/defaultProvider'
+import getObssContract from 'helpers/getObssContract'
+
+export default async function (type: 'vc' | 'founder') {
```
Comment (smyzin) [inline src/helpers/getAllowList.ts]: Use types here instead of `'vc' | 'founder'`

File: src/helpers/fetchAllHashes.ts [hunk 2]
```diff
@@ -17,14 +16,8 @@ function parsePostLogData({
...
-export default async function () {
-  const obssContract = getObssContract()
-
-  const vcAllowMapAddress = await obssContract.vcAllowMap()
-  const vcAllowMap = KetlAllowMap__factory.connect(
-    vcAllowMapAddress,
-    defaultProvider
-  )
+export default async function (type: AllowListType) {
+  const vcAllowMap = await getAllowList(type)
```
Comment (smyzin) [inline src/helpers/fetchAllHashes.ts]: It's better to rename `vcAllowMap` to `allowMap` or smt else

**Review by smyzin — APPROVED** [review]

File: src/helpers/createProof.ts [hunk 3]
```diff
@@ -1,13 +1,27 @@
...
+        'zk/AllowMapChecker.wasm',
+        'zk/AllowMapChecker_final.zkey'
+      )
+      return { proof, type }
+    } catch (e) {
+      console.error(e)
+    }
+  }
+
+  return Promise.reject(new Error(`Can't generate valid proof with this token`))
```
Comment (T-Damer) [inline src/helpers/createProof.ts:29]: Should we return this inside the `catch` block?

**Review by T-Damer — APPROVED** [review]

File: src/helpers/getAllowList.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { AllowListType } from 'models/AllowListType'
+import { KetlAllowMap__factory } from '@big-whale-labs/ketl-allow-map-contract'
+import defaultProvider from 'helpers/defaultProvider'
+import getObssContract from 'helpers/getObssContract'
+
+export default async function (type: AllowListType) {
+  const obssContract = getObssContract()
+
+  const vcAllowMapAddress =
+    type === 'vc'
```
Comment (smyzin) [inline src/helpers/getAllowList.ts]: I would rather use an enum in such cases. But for, lgtm

**Review by smyzin — APPROVED** [review]

File: src/helpers/createProof.ts (hunk 3 above)
Comment (MixailE) [reply src/helpers/createProof.ts:29]: We shouldn't return anything inside loop in `catch` block)

**Pull request merged**